--- target/catkin_ws/src/image_common/camera_calibration_parsers/CMakeLists.txt
+++ target/catkin_ws/src/image_common/camera_calibration_parsers/CMakeLists.txt
@@ -11,11 +11,8 @@
   CATKIN_DEPENDS sensor_msgs
 )
 
-find_package(PkgConfig)
-pkg_check_modules(NEW_YAMLCPP yaml-cpp>=0.5)
-if(NEW_YAMLCPP_FOUND)
+find_package(YamlCpp)
 add_definitions(-DHAVE_NEW_YAMLCPP)
-endif(NEW_YAMLCPP_FOUND)
 
 # define the library
 add_library(${PROJECT_NAME}
