diff --git target/catkin_ws/src/class_loader/CMakeLists.txt target/catkin_ws/src/class_loader/CMakeLists.txt
index da1c8bd..535e0b6 100644
--- target/catkin_ws/src/class_loader/CMakeLists.txt
+++ target/catkin_ws/src/class_loader/CMakeLists.txt
@@ -23,7 +23,7 @@ set(${PROJECT_NAME}_SRCS
   src/meta_object.cpp
   src/multi_library_class_loader.cpp
 )
-add_library(${PROJECT_NAME} SHARED ${${PROJECT_NAME}_SRCS})
+add_library(${PROJECT_NAME} ${${PROJECT_NAME}_SRCS})
 target_link_libraries(${PROJECT_NAME} ${Boost_LIBRARIES} ${console_bridge_LIBRARIES} ${Poco_LIBRARIES})
 
 install(TARGETS ${PROJECT_NAME}
