--- libs/libccd-2.0/CMakeLists.txt	2015-01-14 16:21:25.896887489 -0300
+++ libs/libccd-2.0/CMakeLists.txt	2015-01-14 16:20:44.836888264 -0300
@@ -41,7 +41,7 @@
   src/support.c
   src/vec3.c)
 
-add_library(ccd SHARED ${SOURCE_FILES})
+add_library(ccd STATIC ${SOURCE_FILES})
 set_target_properties(ccd PROPERTIES VERSION ${CCD_VERSION} SOVERSION ${CCD_SOVERSION})
 
 target_link_libraries(ccd ${MATH})
