--- catkin_ws/src/navigation/robot_pose_ekf/CMakeLists.txt	2014-12-23 15:08:45.470144825 -0300
+++ catkin_ws/src/navigation/robot_pose_ekf/CMakeLists.txt	2015-01-09 12:06:24.105071358 -0300
@@ -2,8 +2,15 @@
 project(robot_pose_ekf)
 
 # bfl (Bayesian Filtering Library) is a third party package that uses pkg-config
+if(NOT ANDROID)
 find_package(PkgConfig)
 pkg_check_modules(BFL REQUIRED orocos-bfl)
+else()
+set(BFL_INCLUDE_DIRS "/opt/roscpp_output/target/include/bfl")
+set(BFL_LIBRARY_DIRS "/opt/roscpp_output/target/lib")
+set(BFL_LIBRARY_DIRS "/opt/roscpp_output/target/lib")
+set(BFL_LIBRARIES "/opt/roscpp_output/target/lib/liborocos-bfl.a")
+endif()
 
 # People building this branch of nav within groovy will need this version:
 include_directories(${BFL_INCLUDE_DIRS})
@@ -81,7 +88,7 @@
 ## Tests are failing on OSX for an unknown reason
 include(CMakeDetermineSystem)
 if(CMAKE_SYSTEM_NAME MATCHES "Linux")
-if(CATKIN_ENABLE_TESTING)
+if(CATKIN_ENABLE_TESTING AND NOT ANDROID)
 
 catkin_download_test_data(
   download_data_ekf_test2_indexed.bag
@@ -116,5 +123,5 @@
 add_rostest(${CMAKE_CURRENT_SOURCE_DIR}/test/test_robot_pose_ekf.launch)
 add_rostest(${CMAKE_CURRENT_SOURCE_DIR}/test/test_robot_pose_ekf_zero_covariance.launch)
 
-endif(CATKIN_ENABLE_TESTING)
+endif(CATKIN_ENABLE_TESTING AND NOT ANDROID)
 endif(CMAKE_SYSTEM_NAME MATCHES "Linux")
