--- catkin_ws/src/geometry_experimental/tf2_ros/CMakeLists.txt	2014-12-03 15:25:45.819804934 -0300
+++ catkin_ws/src/geometry_experimental/tf2_ros/CMakeLists.txt	2014-12-03 15:28:10.395801617 -0300
@@ -1,6 +1,10 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(tf2_ros)
 
+if(NOT ANDROID)
+set(TF2_PY tf2_py)
+endif()
+
 find_package(catkin REQUIRED COMPONENTS
   actionlib
   actionlib_msgs
@@ -11,7 +15,7 @@
   rospy
   tf2
   tf2_msgs
-  tf2_py
+  ${TF2_PY}
 )
 find_package(Boost REQUIRED COMPONENTS thread)
 
@@ -29,7 +33,7 @@
   rosgraph
   tf2
   tf2_msgs
-  tf2_py
+  ${TF2_PY}
 )
 
 include_directories(include ${catkin_INCLUDE_DIRS})
@@ -100,7 +104,7 @@
   rostest
   tf2
   tf2_msgs
-  tf2_py
+  ${TF2_PY}
 )
 
 
