--- catkin_ws/src/navigation/voxel_grid/CMakeLists.txt	2014-12-18 23:43:44.237778048 -0300
+++ catkin_ws/src/navigation/voxel_grid/CMakeLists.txt	2014-12-18 23:44:09.953778904 -0300
@@ -21,6 +21,7 @@
 target_link_libraries(voxel_grid ${catkin_LIBRARIES})
 
 install(TARGETS voxel_grid
+  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
   LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
 )
 
@@ -34,4 +35,4 @@
     voxel_grid
     ${catkin_LIBRARIES}
   )
-endif()
\ No newline at end of file
+endif()
